src/io/flutter/view/FlutterView.java

```diff
@@ -132,14 +133,18 @@ private DefaultActionGroup createToolbar(@NotNull ToolWindow toolWindow, @NotNul
     toolbarGroup.add(registerAction(new PerformanceOverlayAction(app)));
     toolbarGroup.addSeparator();
 
-    if (!FlutterSettings.getInstance().isShowHeapDisplay()) {
-      toolbarGroup.add(registerAction(new OpenTimelineViewAction(app)));
-      toolbarGroup.add(registerAction(new OpenObservatoryAction(app)));
-    }
-    else {
-      toolbarGroup.add(new HeapDisplay.ToolbarComponentAction(parentDisposable, app));
-      toolbarGroup.add(new ObservatoryActionGroup(this, app));
+    final boolean isHorizontal = toolWindow.getAnchor().isHorizontal();
+    if (FlutterSettings.getInstance().isShowHeapDisplay() && isHorizontal) {
+      toolbarGroup.add(HeapDisplay.createToolbarAction(parentDisposable, app));
+      toolbarGroup.addSeparator();
+      toolbarGroup.add(FPSDisplay.createToolbarAction(parentDisposable, app));
+      toolbarGroup.addSeparator();
+      //toolbarGroup.add(new ObservatoryActionGroup(this, app));
     }
+
+    toolbarGroup.add(registerAction(new OpenTimelineViewAction(app)));
+    toolbarGroup.add(registerAction(new OpenObservatoryAction(app)));
```

Comment:
Curious why we're losing the group here?  FWIW: I was thinking this would be a good place to hang display configuration options (assuming we have any)

Reply:
generally I was thinking that we'll want the memory related UI and settings together, and the timeline ui and settings together, but that's not really cohesive in this PR - I'll add the group back in here